### ISSUE TITLE
dynamic block in a codepipeline resource's stage block is not expected

### ISSUE THREAD

**Issue by heikki-l** [hashicorp/vscode-terraform#290]
Hi,

I've got a dynamic block inside a stage and the file gets marked with "problems".

```
resource "aws_codepipeline" "pipeline" {
  name     = "pipeline"
  role_arn = aws_iam_role.pipeline_role.arn

  artifact_store {
    location = aws_s3_bucket.pipeline.bucket
    type     = "S3"
  }

  stage {
    name = "Source"

    action {
      category = "Source"
      owner    = "AWS"
      name     = "Source"
      provider = "S3"
      version  = "1"

      output_artifacts = ["src"]

      configuration = {
        S3Bucket    = "some-bucket"
        S3ObjectKey = "build-package.zip"
      }
    }
  }


stage {
    name = "Deploy"

    dynamic "action" {
      for_each = [for project in var.project_list: project]

      content {
        category = "Build"
        owner    = "AWS"
        name     = "${action.value}-ApplyTerraform"
        provider = "CodeBuild"
        version  = "1"

        input_artifacts = ["src"]

        configuration = {
          ProjectName = "project-name"
        }
      }
    }
  }
}
```

**Comment by tenwit** [comment]
That's tf 0.12 code. You don't say what the identified problem is: is it an IDENT mentioning `aws_iam_role.pipeline_role.arn` or similar? If it is, then you're using terraform indexing, which parses 0.11 terraform code. Disable it in your settings and the problem will disappear.

**Comment by fonsecas72** [comment]
Hello, I have the following config and i stil see a warning in dynamic as in the screenshot

```
"terraform.indexing": {
    "enabled": false,
    "liveIndexing": false,
    "delay": 500,
    "exclude": [
        ".terraform/**/*",
        "**/.terraform/**/*"
    ]
},
"terraform.languageServer": {
    "enabled": true,
    "args": []
},
```
![image](https://user-images.githubusercontent.com/1098293/82488496-e478f580-9ad7-11ea-9a66-2040fc79cef8.png)

**Comment by paultyng** [comment]
We just released [v2.0.0-rc.1](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) of the extension. The main features include:

* Added syntax support for 0.12
* Added [terraform-ls](https://github.com/hashicorp/terraform-ls) usage by default (currently on 0.3.0, which offers basic provider code completion)

You can find additional information and specifics in the [release notes](https://github.com/hashicorp/vscode-terraform/releases/tag/v2.0.0-rc.1) and [CHANGELOG](https://github.com/hashicorp/vscode-terraform/blob/master/CHANGELOG.md).

With this release we expect that many of the prior issues and PRs are no longer relevant or have been addressed, and are therefore being closed. If you feel the action taken on an issue or PR is in error, please comment as such and we can figure out the appropriate way to address it.

We plan to add the final 2.0.0 release to the marketplace soon, but are actively seeking your feedback now on the release candidates. You can download the .vsix from the releases page and [manually install it in VS Code](https://code.visualstudio.com/docs/editor/extension-gallery#_install-from-a-vsix) to try it out.
